src/mono/wasm/debugger/BrowserDebugProxy/ValueTypeClass.cs

```diff
@@ -111,16 +112,46 @@ JObject GetFieldWithMetadata(FieldTypeClass field, JObject fieldValue, bool isSt
             }
         }
 
+        public async Task<string> InvokeToStringAsync(MonoSDBHelper sdbAgent, CancellationToken token)
+        {
+            var typeInfo = await sdbAgent.GetTypeInfo(TypeId, token);
+            if (typeInfo == null)
+                return null;
+            if (typeInfo.Name == "object")
+                return null;
+            Microsoft.WebAssembly.Diagnostics.MethodInfo methodInfo = typeInfo.Info.Methods.FirstOrDefault(m => m.Name == "ToString");
+            if (!IsEnum && methodInfo == null)
+                return null;
+            int[] methodIds = await sdbAgent.GetMethodIdsByName(TypeId, "ToString", IsEnum ? BindingFlags.Default : BindingFlags.DeclaredOnly, token);
+            if (methodIds == null)
+                return null;
+            try {
+                foreach (var methodId in methodIds)
+                {
+                    var methodInfoFromRuntime = await sdbAgent.GetMethodInfo(methodId, token);
+                    if (methodInfoFromRuntime.Info.GetParametersInfo().Length > 0)
+                        continue;
+                    var retMethod = await sdbAgent.InvokeMethod(Buffer, methodId, token, "methodRet");
+                    return retMethod["value"]?["value"].Value<string>();
+                }
+            }
+            catch (Exception e)
+            {
+                sdbAgent.logger.LogDebug($"Error while evaluating ToString method: {e}");
+            }
+            return null;
+        }
+
         public async Task<JObject> ToJObject(MonoSDBHelper sdbAgent, bool forDebuggerDisplayAttribute, CancellationToken token)
         {
             string description = className;
             if (ShouldAutoInvokeToString(className) || IsEnum)
```

Comment:
If a type has a DebuggerDisplayAttribute, and has ToString overridden, then what takes precedence?

Reply:
If it's not a ShouldAutoInvode or a Enum, then DebuggerDisplayAttribute is more important.